addons/addon-base-raas-appstream/packages/base-raas-appstream-services/lib/appstream/appstream-sc-service.js

```diff
@@ -111,7 +116,10 @@ class AppStreamScService extends Service {
     return `${uid}-${sessionSuffix}`.replace(/[^\w+=,.@-]+/g, '').slice(0, 32);
   }
 
-  async urlForFirefox(requestContext, { environmentId }) {
+  async getStreamingUrl(requestContext, { environmentId, applicationId }) {
+    const isAppStreamEnabled = this.settings.get(settingKeys.isAppStreamEnabled);
+    if (!isAppStreamEnabled) return undefined;
```

Comment:
I guess we can just throw an error here

Reply:
I think these methods should not throw errors. These are called via plugins and they only contribute to the URL generation if AppStream is enabled. If not, the regular (SC) URLs take effect.

The logic in `addons/addon-base-raas-appstream/packages/base-raas-appstream-services/lib/plugins/env-sc-connection-url-plugin.js` ensures that if AppStream URL returned by this service is `undefined`, then it just returns back the original SC connection info.

Reply:
I see, I think this is related to my original question on an earlier PR where I mentioned that we shouldn't even call these methods if AppStream isn't enabled. Is that hard to achieve?